gclient/src/api/calls.rs

```diff
@@ -204,6 +206,90 @@ impl GearApi {
             .await
     }
 
+    /// Migrates an active program identified by `src_program_id` onto another
+    /// node identified by `dest_node_api` and returns the migrated program
+    /// identifier.
+    pub async fn migrate_program(
+        &self,
+        src_program_id: ProgramId,
+        dest_node_api: &GearApi,
+    ) -> Result<ProgramId> {
+        if dest_node_api.0.api().gprog(src_program_id).await.is_ok() {
+            return Err(Error::ProgramAlreadyExists(
+                src_program_id.as_ref().encode_hex(),
+            ));
+        }
+
+        let dest_program_id = src_program_id;
+
+        // Collect data from the source program
+        let src_free_balance = self.free_balance(src_program_id).await.map_or_else(
+            |e| {
+                if let Error::GearSDK(GsdkError::StorageNotFound) = e {
+                    Ok(0)
+                } else {
+                    Err(e)
+                }
+            },
+            |v| Ok(v),
+        )?;
```

Comment:
Could we use `or_else` here?
```suggestion
        let src_free_balance = self.free_balance(src_program_id).await.or_else(
            |e| {
                if let Error::GearSDK(GsdkError::StorageNotFound) = e {
                    Ok(0)
                } else {
                    Err(e)
                }
            })?;
```